app/resources/admin_api_key_form.html.template

```diff
@@ -14,6 +14,16 @@
   function $(id) {
     return document.getElementById(id);
   }
+  function updateWriteDomain() {
+    var write_field = $('write_permission');
+    var domain_write_field = $('domain_write_permission');
+    if(write_field.checked) {
```

Comment:
This part can be simplified to:

    $('domain_write_permission').disabled = !write_field.checked;